semver/comparator_max.ts

```diff
@@ -1,15 +1,29 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
-import type { Operator, SemVer } from "./types.ts";
+import type { Comparator, Operator, SemVer } from "./types.ts";
 import { ANY, INVALID, MAX } from "./constants.ts";
+import { isComparator } from "./is_comparator.ts";
 
+/**
+ * @deprecated (will be removed in 0.213.0) Use a {@linkcode Comparator} argument instead.
+ */
+export function comparatorMax(semver: SemVer, operator: Operator): SemVer;
```

Comment:
We should still retain the documentation for this overload. Ditto for `comparatorMin()`.